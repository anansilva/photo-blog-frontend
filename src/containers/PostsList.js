import React, { Component } from 'react';
import { connect } from 'react-redux';

export class PostsList extends Component {
  renderPosts() {
    if (!this.props.posts || this.props.posts.length === 0) {
      return 'No Photos';
    } else {
      return this.props.posts.map((post) => {
        return (
          <div className="post-item" key={post.id}>
            {post.photo}
          </div> 
        );
      });
    }
  }

  render() {
    return (
      <div>
        <div>
          <h3>Blog</h3>
          Let's write a post!
        </div>
        {this.renderPosts()}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { posts: state.posts };
 } 

export default connect(mapStateToProps)(PostsList);
