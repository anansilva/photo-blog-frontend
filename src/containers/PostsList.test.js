import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import { PostsList } from './PostsList';
import configureStore from 'redux-mock-store';

describe('Rendering the app', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PostsList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})

describe('Connected React-Redux PostsList', () => {
  let container;

  it('renders welcome message', () => {
    container = shallow(<PostsList />);   
    const welcome = <div><h3>Blog</h3>Let's write a post!</div>;

    expect(container.contains(welcome)).toEqual(true);
  });
  
  it('renders posts when there are posts', () => {
    const posts = [{id: 1, photo: 'photo.jpg'}, {id: 2, photo: 'ruby.png'}]

    container = shallow(<PostsList posts={posts} />);

    expect(container).toMatchSnapshot();
    expect(container.contains('No Photos')).toEqual(false);

  });
  
  it('renders "no photos" when there are no posts', () => {
    const container = shallow(<PostsList />);
    const noPostsMessage = 'No Photos';
    
    expect(container.contains(noPostsMessage)).toEqual(true);
  });
});
