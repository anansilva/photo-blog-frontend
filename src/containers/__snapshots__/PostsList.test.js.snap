// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected React-Redux PostsList renders posts when there are posts 1`] = `ShallowWrapper {}`;
